### DISCUSSION TITLE
Row level security

### DISCUSSION THREAD

**Discussion by linkle69** [supabase/supabase#2377] [Questions] [answered]
I'm trying to insert a new user in my frontend but the row-level security denies it
I have used the policy template that only lets authenticated users insert new rows
but it still refuses to let me insert data even though in frontend it logs that I do have "authenticated" as the role

Any help on this?

![image](https://user-images.githubusercontent.com/50463727/125783758-9a0c134d-71e0-48ea-8fa6-953ad207e540.png)
![image](https://user-images.githubusercontent.com/50463727/125783826-6f583fea-fccf-4143-b9b8-8c836424b306.png)
![image](https://user-images.githubusercontent.com/50463727/125783849-bba69026-b1db-49a4-86d5-969efb752606.png)
![image](https://user-images.githubusercontent.com/50463727/125783877-cf8bbd19-af82-4a6c-a771-7b83b979a33a.png)

**Comment by dshukertjr** [comment, marked as answer]
Hi @link-discord!

Do you have security policy for select? If not, your case might be related to this issue https://github.com/supabase/postgrest-js/issues/127

In short, by default when you insert data into Supabase, you are not only inserting, but selecting the data that you inserted, thus you need both insert and select rights for the inserted data. To fix this, you can either create security policy for select as well, or use `returning: 'minimal'` option like this:
```typescript
await supabase.from('sample').insert({ key: 'value' }, { returning: 'minimal' })
```

**Reply by linkle69** [reply]
The returning option worked for me and solved my issue. Thank you!
